docs/workflows/transitions/preprocess/preprocess/utils.py

```diff
@@ -211,3 +217,24 @@ def patch_empty_predictions(predictions, labels, no_empty_prediction_fields):
     return patched_predictions
 
 
+def filter_away_low_confidence_lines(predictions, field_config):
+    column_names = {
+        label: {column_name for column_name in config['fields']}
+        for label, config in field_config.items() if config.get('type') == 'lines'
+    }
+
+    for prediction in predictions:
+        label = prediction['label']
+        if field_config.get(label, {}).get('type') == 'lines':
+            for index, line in enumerate(prediction['value']):
+                line = filter_by_top1(line, column_names[label])
+                if not line:
+                    continue
+                # column names that are not present in the line counts as 0% confidence
+                line_columns_present = {line_dict['label'] for line_dict in line}
+                line += [{'confidence': 0.0} for _ in column_names[label] - line_columns_present]
+                average_confidence = np.mean([line_dict['confidence'] for line_dict in line])
+                if average_confidence < MINIMUM_AVERAGE_LINE_CONFIDENCE:
+                    prediction['value'][index] = []
```

Comment:
Will this be returned as an empty line, or will it be pruned away before it is returned? 
We want it to be pruned away.